### ISSUE TITLE
Initializer and increment/decrement: 'int  value = a[(i++)];       --value;' -> final int value =a[i++]-1;

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/Spartanizer#1146]
```
    for (final int l = packed.length(); i < l;) {
      int count = packed.charAt(i++), value = packed.charAt(i++);
      --value;
      do
        result[$++] = value++;
        value++;
      while (--count > 0);
    }
```
no need to make sure the variable is not changed later.

**Comment by yossigil** [comment]
Try to use the new mechanisms: 
- examples
- saving of variables in checking
- using variables in inovcation

**Comment by OriRoth** [comment]
Also consider this:
```Java
int x = 1, y = x;
--x;
```

**Comment by OriRoth** [comment]
We may create a more generalized tipper:
```Java
... x ... ;
--x; // or x++
```
==>
```Java
... x++ ... ;
```
It is up to you.

**Comment by yossigil** [comment]
@NivShalmon : good luck.

**Comment by yossigil** [comment]
This should have been done in week1. @ravivos : please manage

**Comment by yossigil** [comment]
I think @NivShalmon pretty much finished this

**Comment by NivShalmon** [comment]
@yossigil This is indeed mostly done, I'm just fixing old tests it broke before pushing this. However, I ran into what I belive is either a bug in `collect#forAllOccurencesExcludingDefinitions`, or me not using it properly. I used it on the list of fragments of the variable declaration statement to see if the variable is used besides being initializer (for the test case ori suggested), however it also added the declaration into the list of occurences (even if it was the only fragment in the list). Should I open a new issue for this? Or is this me using this wrong?
